exporters/zipkin/src/main/java/io/opentelemetry/exporter/zipkin/ZipkinSpanExporter.java

```diff
@@ -143,15 +142,15 @@ private static String nullToEmpty(String value) {
     return value != null ? value : "";
   }
 
-  private static Endpoint chooseEndpoint(SpanData spanData, Endpoint localEndpoint) {
+  private Endpoint getEndpoint(SpanData spanData) {
     Attributes resourceAttributes = spanData.getResource().getAttributes();
 
     // use the service.name from the Resource, if it's been set.
     String serviceNameValue = resourceAttributes.get(ResourceAttributes.SERVICE_NAME);
     if (serviceNameValue == null) {
-      return localEndpoint;
+      serviceNameValue = Resource.getDefault().getAttributes().get(ResourceAttributes.SERVICE_NAME);
     }
-    return Endpoint.newBuilder().serviceName(serviceNameValue).build();
+    return Endpoint.newBuilder().serviceName(serviceNameValue).ip(localAddress).build();
```

Comment:
Is the Endpoint builder re-usable? Could we avoid an allocation or two by keeping a handle to 
`Endpoint.newBuilder().ip(localAddress)` and just tacking on the serviceName each time and building?